sqlitex/pool.go

```diff
@@ -208,6 +208,35 @@ func (p *Pool) Close() (err error) {
 	return
 }
 
+// Size returns the maximum number of potential connections in the Pool.
+// Safe to call on a zero Pool
+func (p *Pool) Size() int {
+	return cap(p.free)
+}
+
+// ExecScript executes a script of SQL statements on all connections.
+// Useful for e.g., applying the same pragma to the entire pool.
+// Each invocation of the script is wrapped in a SAVEPOINT transaction,
+// which is rolled back on any error.
+// But if any invocation errors, the pool is closed to avoid inconsistent behaviour
+// between connections
+func (p *Pool) ExecScript(script string) (err error) {
+	// p.mu.RLock()
+	// defer p.mu.Unlock()
+	conns := make([]*sqlite.Conn, p.Size()) //used to drain the pool
+	for i := 0; i < len(conns); i++ {
+		conns[i] = p.Get(nil)
+		if err = ExecScript(conns[i], script); err != nil {
+			p.Close() //ignoring potential Close() errors
+			return err
+		}
+	}
+	for i := 0; i < len(conns); i++ {
+		p.Put(conns[i])
+	}
+	return nil
+}
```

Comment:
I appreciate the intent of this function but I see some problems with its semantics. For one, the name doesn't properly communicate that the script is executed against all connections in the pool. Additionally since it calls Get p.Size() times, its not obvious that this may block forever if there is a connection not yet returned to the pool. Blocking forever is also a problem. This doesn't expose the context of Get to allow for setting a timeout or for cancellation of the blocking operation. 

For these reasons I can't accept this design as is. I'm going to comment over on issue #101 about what I think a better approach here is for the API.